protoc-gen-swagger/genswagger/template.go

```diff
@@ -329,17 +329,27 @@ func renderEnumerationsAsDefinition(enums enumMap, d swaggerDefinitionsObject, r
 
 // Take in a FQMN or FQEN and return a swagger safe version of the FQMN
 func fullyQualifiedNameToSwaggerName(fqn string, reg *descriptor.Registry) string {
-	return resolveFullyQualifiedNameToSwaggerName(fqn, append(reg.GetAllFQMNs(), reg.GetAllFQENs()...))
+	if lastRegistry != reg {
+		lastRegistryMapping = resolveFullyQualifiedNameToSwaggerNames(append(reg.GetAllFQMNs(), reg.GetAllFQENs()...))
+		lastRegistry = reg
+	}
+	return lastRegistryMapping[fqn]
 }
 
+// lastRegistry is used to remember the last registry passed into fullyQualifiedNameToSwaggerName;
+// we use that to memoise calls to resolveFullyQualifiedNamesToSwaggerNames since it can be
+// significantly slow for registries of hundreds of entries.
+var lastRegistry *descriptor.Registry
```

Comment:
can we instead do something like ```var fqmnCache = map[*descriptor.Registry]map[string]string``` so we can avoid the extra 'last' concept and just compute the cache once per registry seen?